markup.go

```diff
@@ -170,7 +170,19 @@ func If(cond bool, markup ...MarkupOrComponentOrHTML) MarkupOrComponentOrHTML {
 	if cond {
 		return List(markup)
 	}
-	return nil
+	var result []MarkupOrComponentOrHTML
+	for _, m := range markup {
+		if m == nil {
+			continue
+		}
+		switch m.(type) {
+		case *HTML, Component:
+			result = append(result, Tag("noscript"))
+		default:
+			return nil
```

Comment:
Sorry, reading comprehension failure on my part.  I don't really want to return <noscript> tags here at all, this was just a stop-gap change that highlights the problem with the current persistence implementation wrt changing sibling counts.  I must admit I hadn't considered the `Tag()` markup though.